producers/yarn_audit/types/yarn-issue.go

```diff
@@ -174,51 +152,86 @@ type contact struct {
 	Name string `json: name`
 }
 
-// YarnAuditReport includes yarn audit data grouped by advisories, actions and summary
-type YarnAuditReport struct {
-	AuditAdvisories []*auditAdvisoryData
-	AuditActions  []*auditActionData
-	AuditSummary       *auditSummaryData
-}
+func NewReport(report []byte) (*[]AuditAction, *[]AuditAdvisory, *[]AuditSummary, error) {
```

Comment:
Why are we returning pointers instead of just the slices of values?

Reply:
Also should we introduce a report type to wrap all internals?

Reply:
good point - that was an attempt to keep consistent with the previous version of the producer, but probably better removed